addons/website/static/src/scss/secondary_variables.scss

```diff
@@ -14,8 +14,20 @@ $o-color-palettes: append($o-color-palettes, map-merge($-palette-default, o-map-
 
 // Theme color palette
 // -------------------
+//
+// Proper themes have to enforce primary = alpha and secondary = beta.
+// Exceptions to this rule are made for compatibility reasons
+$-palette-default: nth($o-theme-color-palettes, $o-theme-color-palette-number);
+
+$-primary-default: map-get($-palette-default, 'primary');
+$-secondary-default: map-get($-palette-default, 'secondary');
+$-alpha-default: map-get($-palette-default, 'alpha');
+$-beta-default: map-get($-palette-default, 'beta');
 
 // Generate default theme color scheme if alpha is set
+$-is-alpha-primary: not $-primary-default or $-primary-default == $-alpha-default;
+$-is-beta-secondary: not $-secondary-default or $-secondary-default == $-beta-default;
```

Comment:
In fact, I think you don't need to check alpha and beta default right? Just having a "primary" or "secondary" in the default palette is enough to know the theme is a "bad" one for "alpha" or "beta".

Au final: `$-is-alpha-primary: not map-has-key($-palette-default, 'primary')` ?